app/services/api/v3/nodes_stats/response_builder.rb

```diff
@@ -0,0 +1,94 @@
+module Api
+  module V3
+    module NodesStats
+      class ResponseBuilder
+        attr_reader :nodes_stats
+
+        def initialize(commodity_id, contexts_ids, params)
+          @commodity_id = commodity_id
+          @contexts_ids = contexts_ids
+          initialize_params(params)
+          initialize_errors
+        end
+
+        def call
+          initialize_nodes_stats
+          filter_name = @commodity_id ? 'commodity_id' : 'context_id'
+          filter_values =
+            @commodity_id.present? ? [@commodity_id] : @contexts_ids
+          filter_values.map do |filter|
+            {
+              filter_name => filter,
+              attributes: @quants_ids.map do |quant_id|
+                quant = Api::V3::Quant.find(quant_id)
```

Comment:
If the param name is called attribute_ids, these should be ids from attributes_mv, not quants; also, perhaps it would be a bit more consistent to call it attributes_ids (analogically to contexts_ids) and in both cases of attributes_ids and contexts_ids to allow them to be passed as comma-separated strings rather than arrays. I know arrays are more idiomatic in Rails, but at some point we started to switch to comma separated (e.g. in dashboards endpoints) because they're shorter and in some cases that matters.